packages/flutter/lib/src/material/banner.dart

```diff
@@ -386,3 +412,45 @@ class _MaterialBannerState extends State<MaterialBanner> {
     );
   }
 }
+
+class _DefaultsM2 extends MaterialBannerThemeData {
+  _DefaultsM2(this.context)
+    : _theme = Theme.of(context),
+      super(elevation: 0.0);
+
+  final BuildContext context;
+  final ThemeData _theme;
+
+  @override
+  Color? get backgroundColor => _theme.colorScheme.surface;
+
+  @override
+  TextStyle? get contentTextStyle => _theme.textTheme.bodyText2;
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES - Banner
+
+// Do not edit by hand. The code between the "BEGIN GENERATED" and
+// "END GENERATED" comments are generated from data in the Material
+// Design token database by the script:
+//   dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Token database version: v0_101
+
+class _BannerDefaultsM3 extends MaterialBannerThemeData {
+  const _BannerDefaultsM3(this.context)
+    : super(elevation: 1.0);
+
+  final BuildContext context;
+
+  @override
+  Color? get backgroundColor => Theme.of(context).colorScheme.surface;
+
+  @override
+  Color? get surfaceTintColor => Theme.of(context).colorScheme.surfaceTint;
+
+  @override
+  TextStyle? get contentTextStyle => Theme.of(context).textTheme.bodyMedium;
+}
```

Comment:
I see there are more properties (like actions, containerHeight, divider) in the token file. Should we add them in themeData too?

Reply:
Apologies for the delay.

`actions` parameter takes a list of widgets so the styling depends on the widget provided. 

`containerHeight` aka min height isn't customizable currently, there are tokens container height tokens for desktop and mobile. I would prefer to work in a separate PR after Banner is available on https://m3.material.io/.


Added `divider` color and other existing missing properties tests.